### PR TITLE
chore(compiler): add JSDocs to functions relating to component props

### PR DESCRIPTION
**Pull request by alicewriteswrongs**
This documents a few functions and interfaces which have to do with
dealing with component properties, as well as some utils which are used
to creating TypeScript IR for values that we're dealing with at
compile-time.

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Build (`npm run build`) was run locally and any changes were pushed
- [ ] Unit tests (`npm test`) were run locally and passed
- [ ] E2E Tests (`npm run test.karma.prod`) were run locally and passed
- [ ] Prettier (`npm run prettier`) was run locally and passed

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

GitHub Issue Number: N/A

## What is the new behavior?

Most of this is just adding JSDoc comments to functions whose, uh, _function_ I tried to figure out.

One code change is removing a parameter on `parseStaticComponentMeta` which was redundant.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Testing

this should all pass and work and whatnot

### CODE REVIEW

File: src/compiler/transformers/static-to-meta/component.ts [hunk 1]
```diff
@@ -18,14 +18,28 @@ import { parseStringLiteral } from './string-literal';
...
+ * transformation
+ * @returns the TypeScript class declaration instance with which the function
+ * was called
+ */
 export const parseStaticComponentMeta = (
   compilerCtx: d.CompilerCtx,
   typeChecker: ts.TypeChecker,
   cmpNode: ts.ClassDeclaration,
   moduleFile: d.Module,
-  nodeMap: d.NodeMap,
```
Comment (alicewriteswrongs) [inline src/compiler/transformers/static-to-meta/component.ts:26]: the `nodeMap` is already present on `CompilerCtx` and at every call site this function was being called with the `nodeMap` set on the same `CompilerCtx` object already being passed to it, so this argument was redundant and I therefore decided to remove it

**Review by alicewriteswrongs — COMMENTED** [review]
just pointing one thing out

File: src/compiler/transformers/add-component-meta-static.ts [hunk 2]
```diff
@@ -2,7 +2,20 @@ import type * as d from '../../declarations';
 import { convertValueToLiteral, createStaticGetter } from './transform-utils';
 import ts from 'typescript';
 
-export const addComponentMetaStatic = (cmpNode: ts.ClassDeclaration, cmpMeta: d.ComponentCompilerMeta) => {
+/**
+ * Update an instance of TypeScript's IR for a class declaration
```
Comment (rwaskiewicz) [inline src/compiler/transformers/add-component-meta-static.ts:6]: Nit - can we update the first usage of this acronym to include its definition for future readers?
```suggestion
 * Update an instance of TypeScript's Intermediate Representation (IR) for a class declaration
```

File: src/compiler/transformers/static-to-meta/component.ts [hunk 3]
```diff
@@ -18,14 +18,28 @@ import { parseStringLiteral } from './string-literal';
...
 import ts from 'typescript';
 
+/**
+ * Given an instance of typescript's IR for a class decoration ({@see ts.ClassDeclaration})
+ * which represents a Stencil component class declaration, parse and format
+ * various pieces of data about static class members which we use in the
+ * compilation process
+ *
+ * @param compilerCtx the current compiler context
+ * @param typeChecker a typescript typechecker instance
```
Comment (rwaskiewicz) [inline src/compiler/transformers/static-to-meta/component.ts:28]:
```suggestion
 * @param typeChecker a typescript type checker instance
```

File: src/compiler/transformers/static-to-meta/component.ts [hunk 4]
```diff
@@ -18,14 +18,28 @@ import { parseStringLiteral } from './string-literal';
 import { setComponentBuildConditionals } from '../component-build-conditionals';
 import ts from 'typescript';
 
+/**
+ * Given an instance of typescript's IR for a class decoration ({@see ts.ClassDeclaration})
```
Comment (rwaskiewicz) [inline src/compiler/transformers/static-to-meta/component.ts:22]: Nit - can we update the first usage of this acronym to include its definition for future readers?

```suggestion
Given an instance of Typescript's Intermediate Representation (IR) for a class decoration ({@see ts.ClassDeclaration})
```

File: src/compiler/transformers/transform-utils.ts [hunk 5]
```diff
@@ -54,13 +74,38 @@ export const convertValueToLiteral = (val: any, refs: WeakSet<any> = null) => {
...
   });
   return ts.createArrayLiteral(newList);
 };
 
+/**
+ * Convert a JavaScript object (i.e. an object existing at runtime) to its
+ * corresponding TypeScript internal representation ({@see ts.ObjectLiteralExpression})
+ * as an object literal. This function takes an argument holding a `WeakSet` of
+ * references to objects which is used to avoid circular references. Objects
+ * that are converted in this function are added to the the set, and if an
```
Comment (rwaskiewicz) [inline src/compiler/transformers/transform-utils.ts:100]:
```suggestion
 * that are converted in this function are added to the set, and if an
```

File: src/compiler/transformers/transform-utils.ts [hunk 6]
```diff
@@ -23,7 +23,27 @@ export const isMemberPrivate = (member: ts.ClassElement): boolean => {
...
+ * primitive type like `string`, `boolean`, etc or may be an `Object`, `Array`,
+ * etc.
+ *
+ * Note that this function takes a param (`refs`) with a default value,
+ * normally a value should _not_ be passed for this parameter since it is
+ * intended to be used for recursive calls.
+ *
+ * @param val the value to convert
+ * @param refs a set of references, used in recursive calls to avoid
+ * circular references when creating object literal IR instances. **note that
```
Comment (rwaskiewicz) [inline src/compiler/transformers/transform-utils.ts:38]: Same thing for this file - can we expand the first usage of 'IR' for each JSDoc?

**Review by rwaskiewicz — CHANGES_REQUESTED** [review]
Looks good, mostly nitpicks (well, all of them are nitpicks 🙃 )

File: src/compiler/transformers/add-component-meta-static.ts (hunk 2 above)
Comment (alicewriteswrongs) [reply src/compiler/transformers/add-component-meta-static.ts:6]: yeah I think that's a good idea to do across the board 👍

File: src/compiler/transformers/static-to-meta/component.ts (hunk 3 above)
Comment (alicewriteswrongs) [reply src/compiler/transformers/static-to-meta/component.ts:28]: `"type-checker"`? another possibility haha

`"type checker"` sounds good to me

**Comment by alicewriteswrongs** [comment]
@rwaskiewicz ok I updated things to uniformly refer to `IR` and to explain the acronym on first use. also found a few little typos 😱

**Review by rwaskiewicz — APPROVED** [review]
LGTM!

**Pull request merged**